bindings/wasm/src/did/wasm_document.rs

```diff
@@ -139,20 +238,84 @@ impl WasmDocument {
   ///
   /// Throws an error if the method is not found.
   #[wasm_bindgen(js_name = resolveMethod)]
-  pub fn resolve_method(&self, query: &UDIDUrlQuery) -> Result<WasmVerificationMethod> {
+  pub fn resolve_method(&self, query: &UDIDUrlQuery, scope: Option<WasmMethodScope>) -> Result<WasmVerificationMethod> {
     let method_query: String = query.into_serde().wasm_result()?;
+    if let Some(scope) = scope {
+      return Ok(WasmVerificationMethod(
+        self
+          .0
+          .resolve_method_with_scope(&method_query, scope.0)
+          .ok_or(identity::did::Error::MethodNotFound)
+          .wasm_result()?
+          .clone(),
+      ));
+    }
     Ok(WasmVerificationMethod(
       self.0.try_resolve_method(&method_query).wasm_result()?.clone(),
     ))
   }
 
+  /// Attempts to resolve the given method query into a method capable of signing a document update.
+  #[wasm_bindgen(js_name = resolveSigningMethod)]
+  pub fn resolve_signing_method(&mut self, query: &UDIDUrlQuery) -> Result<WasmVerificationMethod> {
+    let method_query: String = query.into_serde().wasm_result()?;
+    Ok(WasmVerificationMethod(
+      self.0.try_resolve_signing_method(&method_query).wasm_result()?.clone(),
+    ))
+  }
+
   #[wasm_bindgen(js_name = revokeMerkleKey)]
   pub fn revoke_merkle_key(&mut self, query: &UDIDUrlQuery, index: u32) -> Result<bool> {
     let method_query: String = query.into_serde().wasm_result()?;
     let method: &mut IotaVerificationMethod = self.0.try_resolve_method_mut(&method_query).wasm_result()?;
     method.revoke_merkle_key(index).wasm_result()
   }
 
+  /// Attaches the relationship to the given method, if the method exists.
+  ///
+  /// Note: The method needs to be in the set of verification methods,
+  /// so it cannot be an embedded one.
+  #[wasm_bindgen(js_name = attachMethodRelationShips)]
+  pub fn attach_method_relationships(&mut self, options: &MethodRelationshipOptions) -> Result<()> {
+    let relationships: Vec<MethodRelationship> = options
+      .relationships()
+      .into_serde::<OneOrMany<WasmMethodRelationship>>()
+      .map(OneOrMany::into_vec)
+      .wasm_result()?
+      .into_iter()
+      .map(MethodRelationship::from)
+      .collect();
+
+    for relationship in relationships {
+      self
+        .0
+        .attach_method_relationship(&options.did_url().0, relationship)
+        .wasm_result()?;
+    }
+    Ok(())
+  }
+
+  /// Detaches the given relationship from the given method, if the method exists.
+  #[wasm_bindgen(js_name = detachMethodRelationShips)]
+  pub fn detach_method_relationships(&mut self, options: &MethodRelationshipOptions) -> Result<()> {
```

Comment:
Nit: casing.

Also same comment as `attachMethodRelationships` regarding the parameters.

```suggestion
  /// Detaches the given relationship from the given method, if the method exists.
  #[wasm_bindgen(js_name = detachMethodRelationships)]
  pub fn detach_method_relationships(&mut self, did_url: &WasmDIDUrl, relationships: &UMethodRelationships) -> Result<()> {
```